File: src/main/java/ir/mmd/intellijDev/Actionable/find/settings/UI.java
package ir.mmd.intellijDev.Actionable.find.settings;

import javax.swing.*;
import java.awt.event.MouseAdapter;
import java.awt.event.MouseEvent;

public class UI {
	private JPanel component;
	private JCheckBox caseSensitiveCheckBox;
	private JButton caseSensitiveCheckBoxDefault;
	
	public UI() {
		initListeners();
	}
	
	private void initListeners() {
		caseSensitiveCheckBoxDefault.addMouseListener(new MouseAdapter() {
			@Override
			public void mouseClicked(MouseEvent e) {
				caseSensitiveCheckBox.setSelected(SettingsState.Defaults.IS_CASE_SENSITIVE);
			}
		});
	}
	
	public JPanel getComponent() {
		return component;
	}
	
	public boolean isCaseSensitive() {
		return caseSensitiveCheckBox.isSelected();
	}
	
	public void setCaseSensitive(boolean b) {
		caseSensitiveCheckBox.setSelected(b);
	}
}

File: src/main/java/ir/mmd/intellijDev/Actionable/find/settings/UI.kt
package ir.mmd.intellijDev.Actionable.find.settings

import com.intellij.ui.dsl.builder.*
import ir.mmd.intellijDev.Actionable.ActionableBundle
import ir.mmd.intellijDev.Actionable.util.observableMutablePropertyOf

class UI {
	private val isCaseSensitiveProperty = observableMutablePropertyOf(SettingsState.Defaults.isCaseSensitive)
	var isCaseSensitive by isCaseSensitiveProperty
	
	val component = panel {
		row {
			checkBox(ActionableBundle.string("findPanel.caseSensitiveCheckBox.label"))
				.bindSelected(isCaseSensitiveProperty)
				.align(Align.FILL)
				.comment(ActionableBundle.string("findPanel.caseSensitiveCheckBox.comment"))
			
			button(ActionableBundle.string("global.defaultButton.label")) {
				isCaseSensitive = SettingsState.Defaults.isCaseSensitive
			}.align(AlignX.RIGHT + AlignY.CENTER)
		}
	}
}
